svm/src/transaction_processor.rs

```diff
@@ -734,57 +739,55 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
     fn validate_transaction_nonce<CB: TransactionProcessingCallback>(
         account_loader: &mut AccountLoader<CB>,
         message: &impl SVMMessage,
-        nonce_info: &NonceInfo,
+        nonce_address: &Pubkey,
         next_durable_nonce: &DurableNonce,
+        next_lamports_per_signature: u64,
         error_counters: &mut TransactionErrorMetrics,
-    ) -> TransactionResult<()> {
+    ) -> TransactionResult<NonceInfo> {
         // When SIMD83 is enabled, if the nonce has been used in this batch already, we must drop
         // the transaction. This is the same as if it was used in different batches in the same slot.
-        // If the nonce account was closed in the batch, we error as if the blockhash didn't validate.
-        // We must validate the account in case it was reopened, either as a normal system account,
-        // or a fake nonce account. We must also check the signer in case the authority was changed.
-        //
-        // Note these checks are *not* obviated by fee-only transactions.
-        let nonce_is_valid = account_loader
-            .load_transaction_account(nonce_info.address(), true)
-            .and_then(|ref current_nonce| {
-                system_program::check_id(current_nonce.account.owner()).then_some(())?;
-                StateMut::<NonceVersions>::state(&current_nonce.account).ok()
-            })
-            .and_then(
-                |current_nonce_versions| match current_nonce_versions.state() {
-                    NonceState::Initialized(current_nonce_data) => {
-                        let nonce_can_be_advanced =
-                            &current_nonce_data.durable_nonce != next_durable_nonce;
-
-                        let nonce_authority_is_valid = message
-                            .account_keys()
-                            .iter()
-                            .enumerate()
-                            .any(|(i, address)| {
-                                address == &current_nonce_data.authority && message.is_signer(i)
-                            });
-
-                        if nonce_authority_is_valid {
-                            Some(nonce_can_be_advanced)
-                        } else {
-                            None
-                        }
-                    }
-                    _ => None,
-                },
+        // It is possible that the nonce account was used, closed, closed and reopened, closed and
+        // spoofed by a non-system program, or had its authority changed. Such a transaction cannot
+        // be processed, even as fee-only.
+
+        let Some(mut nonce_account) = account_loader
+            .load_transaction_account(nonce_address, true)
+            .map(|loaded| loaded.account)
+        else {
+            error_counters.account_not_found += 1;
+            return Err(TransactionError::AccountNotFound);
```

Comment:
I'm happy with this change but can you let FD know that the error for this case will change from `BlockhashNotFound` to `AccountNotFound` so they can match this change?